### ISSUE TITLE
Deblur-NeRF: Neural Radiance Fields from Blurry Images

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#166]
## TL; DR

- Learn additional spatially dependent blur kernel so that NeRF can be trained with blurry images.

![image](https://user-images.githubusercontent.com/46308407/162126508-5e13f3e8-7d68-4939-9d72-85144b163e89.png)

## Motivation

- We cannot always expect our training data to be clean and perfectly aligned.

## Prior works

- NeRF
- Image Deblurring
- Camera Response Function (intensity - irradiance)

## Method

![image](https://user-images.githubusercontent.com/46308407/162126615-1fd65e92-018e-4318-bfb0-41dd3b3217c5.png)

- There can be many types of blurring, but to name the two most representative, they are *camera motion blur* and *defocus blur*.

![image](https://user-images.githubusercontent.com/46308407/162126534-74654113-4d07-4c0d-bd74-8eafab29feb8.png)

- For each query point in 3D coordinate and the viewing direction, obtain blurring kernel as well as ray origin from a learnable MLP.

![image](https://user-images.githubusercontent.com/46308407/162128034-9f34c28d-31a2-4f84-8f30-91e4d563a062.png)

- Input the transported ray origin to the NeRF (designed to output *sharp* colors) and apply blurring kernel (weights = output from previous MLP) for blur adjustment.

![image](https://user-images.githubusercontent.com/46308407/162128105-2a0ceb6e-9fd6-4d71-9fbc-4f5f26b3fd08.png)

- Instead of directly working with intensity, model irradiance and adapt it using gamma correction function.

![image](https://user-images.githubusercontent.com/46308407/162128159-4bd8f921-9e9b-4470-b2d2-6bc87a2a8ba5.png)

- Train with pixel reconstruction objective similar to NeRF after blurring operation. *(GT is blurred)*

![image](https://user-images.githubusercontent.com/46308407/162128074-4609ae8f-db3a-4753-b0f2-8effab6e578f.png)

- Additional regularization terms boost performance.

![image](https://user-images.githubusercontent.com/46308407/162128196-82e5df67-86d1-48d1-bd67-6a55a43d8850.png)

- This enforces that q0, a fixed point in the pixel neighborhood, to be at the center of the blur kernel.
- Also the ray origin should not change to much as to destabilize training.

### Ablations

![image](https://user-images.githubusercontent.com/46308407/162128440-ac883d72-f6e5-40af-9f18-932a31f65304.png)

![image](https://user-images.githubusercontent.com/46308407/162128461-42f29f68-e89a-4f1e-872a-674476e6e01d.png)

### Baseline comparison

![image](https://user-images.githubusercontent.com/46308407/162128506-3db00c01-4627-4fb7-9f87-0c0dea58c633.png)
![image](https://user-images.githubusercontent.com/46308407/162128517-ebda83a4-47f0-4f45-85c5-dd82386d3970.png)

## Comments

- As the authors admit in their discussion section, the problem of learning accurate 3D geometry from a set of blurred images is *ill-posed*, providing no guarantee as to the desired disentanglement (blurring kernel and sharp geometry)
- However, in my opinion, there is an underlying implicit constraint of ***actual 3D geometry of the scene***, which guides the model to learn the fixed geometry.
- In a sense, NeRF has been shown to be extremely good at capturing 3D geometry from a set of posed images, so perhaps having some noise in the observations might not *fundamentally* change the problem formulation.
- In practical sense, this is equivalent to data preprocessing, since NeRF variants are not designed to *generalize*. If we can deblur the training images and provide accurate camera annotations, vanilla NeRF will have no problem at all. Quantitative comparison with deblur-then-NeRF methods might be a little misleading because the performance drop could be (at least partially) attributed to inaccurate camera annoations coming from blurring (*i.e., camera motion blur*).
